substrate/runtime/executive/src/lib.rs

```diff
@@ -218,6 +218,11 @@ impl<
 		let storage_root = System::Hashing::storage_root();
 		header.state_root().check_equal(&storage_root);
 		assert!(header.state_root() == &storage_root, "Storage root must match that calculated.");
+
+		// check storage changes root
+		let storage_changes_root = System::Hashing::storage_changes_root();
+		header.changes_root().cloned().check_equal(&storage_changes_root);
```

Comment:
Why do we need `cloned` here?

Reply:
Because `CheckEqual` was implemented for `Option<T>`. Changed to `for Option<&T>`.